Sources/Widgets/Widgets3D/ResliceCursorWidget/behavior.js

```diff
@@ -83,36 +85,93 @@ export default function widgetBehavior(publicAPI, model) {
+    const ZPlaneNormal = model.widgetState.getZPlaneNormal();
+
+    // Find in which view we are
+    let otherViews = ['X', 'Y'];
+    let otherNormals = [XPlaneNormal, YPlaneNormal];
+    if (vtkMath.dot(normal, XPlaneNormal) > 0.9) {
+      otherViews = ['Y', 'Z'];
+      otherNormals = [YPlaneNormal, ZPlaneNormal];
+    } else if (vtkMath.dot(normal, YPlaneNormal) > 0.9) {
+      otherViews = ['X', 'Z'];
+      otherNormals = [XPlaneNormal, ZPlaneNormal];
+    }
 
+    // Define the potentially new center
     let newCenter = [
-      currentCenter[0] + direction * translationVector[0] * imageSpacing[0],
-      currentCenter[1] + direction * translationVector[1] * imageSpacing[1],
-      currentCenter[2] + direction * translationVector[2] * imageSpacing[2],
+      currentCenter[0] + direction * normal[0] * imageSpacing[0],
+      currentCenter[1] + direction * normal[1] * imageSpacing[1],
+      currentCenter[2] + direction * normal[2] * imageSpacing[2],
     ];
-
-    if (!boundingBox.containsPoint(...newCenter)) {
-      // Clamp the center to the bounds of the image
-      const imageOrigin = image.getOrigin();
-      const localRoundedCenter = currentCenter.map(
-        (value, index) =>
-          Math.round(value - imageOrigin[index]) / imageSpacing[index]
-      );
-
-      // Get it back in world coordinates
-      newCenter = localRoundedCenter.map(
-        (value, index) => value * imageSpacing[index] + imageOrigin[index]
-      );
+    const newCenterInsideBounds = bounds.containsPoint(...newCenter);
+
+    // The center is out of the image
+    if (!newCenterInsideBounds) {
+      // Find which plane is out of the bounds
+      const isCurrentPlaneOutOfBounds =
+        !vtkBox.intersectPlane(imageBounds, newCenter, normal) &&
+        vtkBox.intersectPlane(imageBounds, currentCenter, normal);
+
+      const otherViewsPlaneOutOfBounds = [];
+      let otherPlanesAreOutsideBounds = false;
+      for (let i = 0; i < otherViews.length; i++) {
+        const outBounds =
+          !vtkBox.intersectPlane(imageBounds, newCenter, otherNormals[i]) &&
+          vtkBox.intersectPlane(imageBounds, currentCenter, otherNormals[i]);
+        otherViewsPlaneOutOfBounds.push(outBounds);
+        otherPlanesAreOutsideBounds = otherPlanesAreOutsideBounds && outBounds;
+      }
+
+      // Basic clamping of the newCenter to the image bounds
+      if (!otherPlanesAreOutsideBounds && isCurrentPlaneOutOfBounds) {
+        // Find new reslice cursor center in local coordinates and round it to be on a valid slice
+        const imageOrigin = image.getOrigin();
+        const localRoundedCenter = currentCenter.map(
+          (value, index) =>
+            Math.round(value - imageOrigin[index]) / imageSpacing[index]
+        );
+
+        // Get it back in world coordinates
+        newCenter = localRoundedCenter.map(
+          (value, index) => value * imageSpacing[index] + imageOrigin[index]
```

Comment:
if image origin is 0,0,0, spacing (1,1,1) and dimensions( 10, 10, 10).

if currentCenter is (5,5,5) and newCenter is (15, 5, 5), then localRoundedCenter would be 5, 5, 5 and newCenter (5, 5, 5) but we want (10, 5, 5).

Reply:
Indeed. That's a mistake. It's only use to get the correct id of a slice.

Reply:
As we are moving spacing by spacing with the mouse wheel, these code only clamped the value to the closest slice.
Following your example, if the currentCenter is (9.5, 5, 5), then after using the mouse wheel, you'll have (10.5, 5, 5). This code will clamp it to (10, 5, 5)

Reply:
I think for the sake of genericity, it should not be assumed that you are moving spacing by spacing.

Reply:
(as a side note, if currentCenter is (9.1, 5, 5) then after a mouse wheel of 1, you'll have (10.5, 5, 5). The code will clamp it to (9, 5, 5) and you will therefore never be able to see the slice 10)

Reply:
btw, if middle button is down and moved, then it should scroll (not just mouse wheel).